shell/egl_window.h

```diff
@@ -107,25 +120,22 @@ class EglWindow : public Egl {
                                int height,
                                uint32_t format);
 
+  static const struct xdg_surface_listener xdg_surface_listener;
+
   static void handle_xdg_surface_configure(void* data,
                                            struct xdg_surface* xdg_surface,
                                            uint32_t serial);
 
-  static const struct xdg_surface_listener xdg_surface_listener;
-
-  static const struct wl_callback_listener xdg_configure_callback_listener;
-
-#if HIDPI_CLIENT
-  static const struct wl_surface_listener primary_surface_listener;
+  static const struct xdg_toplevel_listener xdg_toplevel_listener;
 
-  static void handle_primary_surface_enter(void* data,
-                                           struct wl_surface* wl_surface,
-                                           struct wl_output* output);
+  static void handle_toplevel_configure(void* data,
+                                        struct xdg_toplevel* toplevel,
+                                        int32_t width,
+                                        int32_t height,
+                                        struct wl_array* states);
 
-  static void handle_primary_surface_leave(void* data,
-                                           struct wl_surface* wl_surface,
-                                           struct wl_output* output);
-#endif
+  static void handle_toplevel_close(void* data,
```

Comment:
All of the toplevel bits need to go in the first patch, added here need to go in the first patch.